cpp/source/deal.II/mp_values.h

```diff
@@ -101,6 +93,26 @@ class UniformConstantMPValues : public MPValues<dim>
 };
 
 template <int dim>
+class FunctionSpaceMPValues : public MPValues<dim>
+{
+public:
+  FunctionSpaceMPValues(std::shared_ptr<dealii::Function<dim>> const &function);
+  FunctionSpaceMPValues(boost::property_tree::ptree const &ptree);
+
+  /*!
+    \note
+    Here \p fe_values must be constructed with the flag \c
+    dealii::update_quadrature_points.
+  */
+  void get_values(std::string const &key,
+                  dealii::FEValues<dim> const &fe_values,
+                  std::vector<double> &values) const override;
+
+private:
+  std::shared_ptr<dealii::Function<dim>> const _function;
```

Comment:
more general.  we could consider move uniform constant to simple `dealii::ConstantFunction<dim>`